src/iframe/storage/unpartitioned-storage/local-storage.ts

```diff
@@ -42,4 +89,536 @@ export class LocalStorage extends EnhancedStorage {
+        return cookie?.value || null;
+      } else {
+        return Cookies.get(key) || null;
+      }
+    } catch (e) {
+      if (LocalStorage.DEBUG_MODE) console.warn(`Failed to get cookie for key '${key}'`, e);
+
+      /*
+      try {
+        return Cookies.get(key) || null;
+      } catch (fallbackError) {
+        if (LocalStorage.DEBUG_MODE) console.warn(`Fallback cookie get also failed for key '${key}'`, fallbackError);
+      }
+      */
+
+      return null;
+    }
+  }
+
+  /**
+   * Get all cookies using CookieStore API or js-cookie fallback
+   */
+  private async getAllCookies(): Promise<Record<string, string>> {
+    try {
+      const cookieStore = this.getCookieStore();
+      if (cookieStore) {
+        const cookies = await cookieStore.getAll();
+        return cookies.reduce(
+          (acc, cookie) => {
+            acc[cookie.name] = cookie.value;
+            return acc;
+          },
+          {} as Record<string, string>,
+        );
+      } else {
+        return Cookies.get() || {};
+      }
+    } catch (e) {
+      if (LocalStorage.DEBUG_MODE) console.warn(`Failed to get all cookies`, e);
+
+      /*
+      try {
+        return Cookies.get() || {};
+      } catch (fallbackError) {
+        if (LocalStorage.DEBUG_MODE) console.warn(`Fallback getAll cookies also failed`, fallbackError);
+      }
+      */
+
+      return {};
+    }
+  }
+
+  private async getAllStorageCookieKeys(): Promise<string[]> {
+    const allCookies = await this.getAllCookies();
+
+    return Object.keys(allCookies).filter(
+      (key) => this.shouldStoreInCookies(key) || key.includes(LocalStorage.CHUNK_SUFFIX),
+    );
+  }
+
+  /**
+   * Get a value from storage with appropriate method
+   */
+  private getStorageValue<T = string>(key: string, useRaw = false): T | null {
+    try {
+      return useRaw ? (this.storage.getRaw(key) as T) : (this.storage.getItem(key) as T);
+    } catch (e) {
+      if (LocalStorage.DEBUG_MODE) console.warn(`Failed to get storage value for key '${key}'`, e);
+      return null;
+    }
+  }
+
+  /**
+   * Set a value to storage with appropriate method
+   */
+  private setStorageValue<T>(key: string, value: T, useRaw = false): boolean {
+    try {
+      if (useRaw) {
+        this.storage.setRaw(key, value as string);
+      } else {
+        this.storage.setItem(key, value);
+      }
+      return true;
+    } catch (e) {
+      if (LocalStorage.DEBUG_MODE) console.warn(`Failed to set storage value for key '${key}'`, e);
+      return false;
+    }
+  }
+
+  /**
+   * Set a cookie with chunking if needed
+   */
+  private async setCookieWithChunkingIfNeeded(key: string, value: string): Promise<boolean> {
+    if (!value) {
+      if (LocalStorage.DEBUG_MODE) console.warn(`Attempted to set empty value for key '${key}'`);
+      return false;
+    }
+
+    // Check if value is too large for a single cookie
+    if (value.length > LocalStorage.MAX_COOKIE_CHUNK_SIZE) {
```

Comment:
I think some browsers do allow for way larger cookies, so trying to set it in a `try..catch` block is probably a better approach.